### ISSUE TITLE
Image Parameters

### ISSUE THREAD

**Issue by heyjoecampbell** [joomlaworks/k2-v3-dev-build#220]
Offer image parameter(s) to control the processing of all images:

- Retain File Name: K2 could add the md5 hash to beginning of file name and size at end of file name while retaining keywords in the middle portion of the file name (ea457adccaa9e569cff05de9b4f3b04d-My-Original-File-Name-XL.jpg) Also: only use dashes instead of underscores

- Extra Field Image Output: allow user to select (enable or disable HTML output) to offer the ability of file path only extra field output

- Image Quality & Size: offer a parameter to upload any images without processing other than file name in order to retain desired quality & compression

**Comment by simbus82** [comment]
Retain file name is one of the smart things that it can be done in this Joomla Component.

Google Images for example check a correspondence between file name and alt tag, and give to image more "power" related to the search term.
The semantic engine see file name, alt tag, anchor text and correspondence with page Headers and Title that contain images.

Google Official
https://support.google.com/webmasters/answer/114016?hl=en

Neil Patel 
http://www.quicksprout.com/2012/11/05/image-optimization-102-the-advance-guide-to-seo-for-images/

Shopify blog
http://www.shopify.com/blog/7412852-10-must-know-image-optimization-tips

Yoast
https://yoast.com/image-seo/

KissMetrics
https://blog.kissmetrics.com/kissmetrics-guide-to-blogging/

RavenTools
http://blog.raventools.com/alt-title-text-optimized-images/

Every SEO in the world know that the file name of an image is the first thing that a search engine consider after the alt attribute.

Leave the file name untouched with only the add of an uniqueId and a tag for image size, is a practice that have NO COST in writing code and do a lot of good work for SEO.

**Comment by heyjoecampbell** [comment]
Yes - that is why I use Extra Fields to render ALL of my K2 images (category, tag, item, related, Open Graph, Twitter Card, etc.), as doing such offers me FULL control of each image (name, quality and file size)
